### PR TITLE
[DEV-3627] Reuse sampled data when executing describe queries

### PR DESCRIPTION
**Pull request by YS-L**
## Description

This updates describe queries to sample input data once and then reuse it.

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

**Review by gee-senbong — APPROVED** [review]

File: featurebyte/service/preview.py [hunk 1]
```diff
@@ -256,21 +256,32 @@ async def describe(
...
-            columns = describe_query.columns
-            assert result is not None
-            df_query = pd.DataFrame(
-                result.values.reshape(len(columns), -1).T,
-                index=describe_query.row_names,
-                columns=[str(column.name) for column in columns],
+        await session.create_table_as(
+            table_details=describe_queries.data.output_table_name,
+            select_expr=describe_queries.data.expr,
+        )
```
Comment (gee-senbong) [inline featurebyte/service/preview.py:262]: Seems like there is an issue when writing invalid dates into spark delta table:
```
CREATE TABLE `__TEMP_SAMPLED_DATA_66A25AD002AF101FA4F82384`
USING DELTA
TBLPROPERTIES (
  'delta.columnMapping.mode'='name',
  'delta.minReaderVersion'='2',
  'delta.minWriterVersion'='5'
) AS
SELECT
  `date_col` AS `date_col`
FROM `spark_catalog`.`featurebyte_20240725220133_999912`.`TABLE_INVALID_DATES`
```
Should we skip `tests.integration.api.test_view_describe.test_describe_invalid_dates` for spark?

Comment (gee-senbong) [reply featurebyte/service/preview.py:262]: If we pass `kind="VIEW"`, it works for spark.

Comment (gee-senbong) [reply featurebyte/service/preview.py:262]: Create a branch to test: https://github.com/featurebyte/featurebyte/pull/2479

Comment (YS-L) [reply featurebyte/service/preview.py:262]: Setting `spark.sql.parquet.int96RebaseModeInWrite=CORRECTED` according to the error message also seems to fix it.

Comment (YS-L) [reply featurebyte/service/preview.py:262]: I think we want to create a table but not a view to ensure the sampling is done once.

**Pull request merged**